src/main/scala/firrtl/ir/IR.scala

```diff
@@ -607,8 +608,16 @@ case class Stop(info: Info, ret: Int, clk: Expression, en: Expression)
   def foreachType(f:   Type => Unit):   Unit = ()
   def foreachString(f: String => Unit): Unit = ()
   def foreachInfo(f:   Info => Unit):   Unit = f(info)
+  def copy(info: Info = info, ret: Int = ret, clk: Expression = clk, en: Expression = en): Stop = {
+    Stop(info, ret, clk, en)
+  }
+}
+object Stop {
+  def unapply(s: Stop): Option[(Info, Int, Expression, Expression)] = {
```

Comment:
```suggestion
  def unapply(s: Stop): Some[(Info, Int, Expression, Expression)] = {
```

This is not something I knew until just today I happened to see a conversation about it on Gitter. In Scala 2.13.4 you'll get an exhaustivity warning unless you return the more specific type: See https://github.com/scodec/scodec/pull/296 and the linked Gitter conversation.